spec/datadog/opentelemetry_spec.rb

```diff
@@ -339,18 +339,21 @@
       describe '#inject' do
         subject(:inject) { ::OpenTelemetry.propagation.inject(carrier) }
         let(:carrier) { {} }
-
+        let(:headers) do
```

Comment:
This is fine, but as a reminder. `let` would cache the result when first invoked, this could potentially cause stale values within a test life cycle if not paying attention, especially the data depends on a variable `Datadog::Tracing.active_trace` that changes with context.

I recommend 

```
def headers
  {
    ...
  }
end
```

Reply:
Good callout, updated the test.